<launch>

  <arg name="experiment_type" default="mapping"/>
  <arg name="export_mesh" default="true"/>
  <arg name="experiments_dir"
      default="/media/scratch1/albanesg/mapping_experiments/mask2former_swin_tiny_scannet"/>
  <arg name="ground_truth_dir" default="/media/scratch1/albanesg/scans"/>

  <node pkg="active_panoptic_mapping_utils" name="evaluation_manager" type="evaluation_manager.py" required="true">
      <param name="experiments_dir" value="$(arg experiments_dir)"/>
      <param name="experiment_type" value="$(arg experiment_type)"/>
      <param name="ground_truth_dir" value="$(arg ground_truth_dir)"/>
  </node>

  <node name="evaluation_data_export" pkg="active_panoptic_mapping_utils" type="evaluation_data_export_node">
    <param name="export_mesh" value="$(arg export_mesh)"/>
    <param name="is_single_tsdf" value="true"/>
    <param name="export_mesh" value="$(arg export_mesh)"/>
    <param name="refine_alignment" value="true" if="$(eval arg('experiment_type') == 'planning')"/>
    <!-- Only required for planning experiments (on Replica for now) -->
    <rosparam param="alignment_transformation">
      [0.0, -1.0, 0.0, 0.0,
       1.0, 0.0, 0.0, 0.0
       0.0, 0.0, 1.0, 0.0,
       0.0, 0.0, 0.0, 1.0]
    </rosparam>
  </node>

</launch>
